### ISSUE TITLE
Floating Point Error of SMCPP split

### ISSUE THREAD

**Issue by biozzq** [popgenmethods/smcpp#25]
Dear @terhorst 

I ran split history using the latest SMCPP (version 1.9), but it gave me an error when running split,

`File "/bin/smcpp/lib/python3.5/site-packages/smcpp/analysis.py", line 93, in _load_data`
    `self._esfs.astype('float') / self._esfs.sum())`
`FloatingPointError: invalid value encountered in double_scalars`

The following is `.debug.txt` which recorded the command lines and running processing.
```
560 smcpp.logging DEBUG SMC++ v1.9.0
561 smcpp.commands.command DEBUG ['/bin/smcpp/bin/smc++', 'split', '-o', 'CNN_CNS', '../CNN/CNN/model.final.json', '../CNS/CNS/model.final.jso
561 smcpp.commands.command DEBUG Namespace(algorithm='L-BFGS-B', blocks=None, command='split', data=['CNN_CNS.10.smc.gz', 'CNS.CNN.10.smc.gz'], em_iter
561 smcpp.analysis DEBUG Polarization error p=0.500000
561 smcpp.analysis DEBUG regularization degree=2
561 smcpp.analysis INFO Loading data...
3373 smcpp.analysis INFO 0.21 Gb of data
3373 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.10.smc.gz', n=array([38, 38]), a=array([2, 0]))
3374 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.10.smc.gz', n=array([36, 40]), a=array([2, 0]))
3374 smcpp.analysis INFO 2 populations
```

Can you help me to fix this error? Thanks.

Best
zhuqing

**Comment by terhorst** [comment]
Thanks for reporting the bug. Would you mind posting your ``.debug.txt`` after updating to v1.9.1 and re-running? I'm not sure why you encountered this (though it should not be fixed).

**Comment by biozzq** [comment]
Dear @terhorst 

It gave me a different error like following,
```
549 smcpp.analysis INFO Loading data...
13600 smcpp.analysis INFO 5.02 Gb of data
13624 smcpp.analysis INFO 2 populations
Traceback (most recent call last):
  File "/bin/smcpp/bin/smc++", line 11, in <module>
    load_entry_point('smcpp==1.9.1', 'console_scripts', 'smc++')()
  File "/bin/smcpp/lib/python3.5/site-packages/smcpp/frontend/console.py", line 23, in main
    cmds[args.command].main(args)
  File "/bin/smcpp/lib/python3.5/site-packages/smcpp/commands/split.py", line 44, in main
    analysis = SplitAnalysis(args.data, args)
  File "/bin/smcpp/lib/python3.5/site-packages/smcpp/analysis.py", line 398, in __init__
    BaseAnalysis.__init__(self, files, args)
  File "/bin/smcpp/lib/python3.5/site-packages/smcpp/analysis.py", line 45, in __init__
    self._load_data(files)
  File "/bin/smcpp/lib/python3.5/site-packages/smcpp/analysis.py", line 93, in _load_data
    sfs = util.undistinguished_sfs(self._esfs)
  File "/bin/smcpp/lib/python3.5/site-packages/smcpp/util.py", line 50, in undistinguished_sfs
    a = sfs.shape[0] - 1
IndexError: tuple index out of range
```

```
549 smcpp.logging DEBUG SMC++ v1.9.1
549 smcpp.commands.command DEBUG ['/bin/smcpp/bin/smc++', 'split', '-o', 'CNN_CNS', '--em-iterations', '20', '../CNN/CNN/model.final.json', '../CNS/CNS/model.final.json', 'CNN_CNS.10.smc.gz', 'CNN_CNS.11.smc.gz', 'CNN_CNS.12.smc.gz', 'CNN_CNS.13.smc.gz', 'CNN_CNS.14.smc.gz', 'CNN_CNS.15.smc.gz', 'CNN_CNS.16.smc.gz', 'CNN_CNS.17.smc.gz', 'CNN_CNS.18.smc.gz', 'CNN_CNS.19.smc.gz', 'CNN_CNS.1.smc.gz', 'CNN_CNS.20.smc.gz', 'CNN_CNS.21.smc.gz', 'CNN_CNS.22.smc.gz', 'CNN_CNS.23.smc.gz', 'CNN_CNS.24.smc.gz', 'CNN_CNS.25.smc.gz', 'CNN_CNS.26.smc.gz', 'CNN_CNS.27.smc.gz', 'CNN_CNS.28.smc.gz', 'CNN_CNS.29.smc.gz', 'CNN_CNS.2.smc.gz', 'CNN_CNS.3.smc.gz', 'CNN_CNS.4.smc.gz', 'CNN_CNS.5.smc.gz', 'CNN_CNS.6.smc.gz', 'CNN_CNS.7.smc.gz', 'CNN_CNS.8.smc.gz', 'CNN_CNS.9.smc.gz', 'CNS.CNN.10.smc.gz', 'CNS.CNN.11.smc.gz', 'CNS.CNN.12.smc.gz', 'CNS.CNN.13.smc.gz', 'CNS.CNN.14.smc.gz', 'CNS.CNN.15.smc.gz', 'CNS.CNN.16.smc.gz', 'CNS.CNN.17.smc.gz', 'CNS.CNN.18.smc.gz', 'CNS.CNN.19.smc.gz', 'CNS.CNN.1.smc.gz', 'CNS.CNN.20.smc.gz', 'CNS.CNN.21.smc.gz', 'CNS.CNN.22.smc.gz', 'CNS.CNN.23.smc.gz', 'CNS.CNN.24.smc.gz', 'CNS.CNN.25.smc.gz', 'CNS.CNN.26.smc.gz', 'CNS.CNN.27.smc.gz', 'CNS.CNN.28.smc.gz', 'CNS.CNN.29.smc.gz', 'CNS.CNN.2.smc.gz', 'CNS.CNN.3.smc.gz', 'CNS.CNN.4.smc.gz', 'CNS.CNN.5.smc.gz', 'CNS.CNN.6.smc.gz', 'CNS.CNN.7.smc.gz', 'CNS.CNN.8.smc.gz', 'CNS.CNN.9.smc.gz']
549 smcpp.commands.command DEBUG Namespace(algorithm='L-BFGS-B', blocks=None, command='split', data=['CNN_CNS.10.smc.gz', 'CNN_CNS.11.smc.gz', 'CNN_CNS.12.smc.gz', 'CNN_CNS.13.smc.gz', 'CNN_CNS.14.smc.gz', 'CNN_CNS.15.smc.gz', 'CNN_CNS.16.smc.gz', 'CNN_CNS.17.smc.gz', 'CNN_CNS.18.smc.gz', 'CNN_CNS.19.smc.gz', 'CNN_CNS.1.smc.gz', 'CNN_CNS.20.smc.gz', 'CNN_CNS.21.smc.gz', 'CNN_CNS.22.smc.gz', 'CNN_CNS.23.smc.gz', 'CNN_CNS.24.smc.gz', 'CNN_CNS.25.smc.gz', 'CNN_CNS.26.smc.gz', 'CNN_CNS.27.smc.gz', 'CNN_CNS.28.smc.gz', 'CNN_CNS.29.smc.gz', 'CNN_CNS.2.smc.gz', 'CNN_CNS.3.smc.gz', 'CNN_CNS.4.smc.gz', 'CNN_CNS.5.smc.gz', 'CNN_CNS.6.smc.gz', 'CNN_CNS.7.smc.gz', 'CNN_CNS.8.smc.gz', 'CNN_CNS.9.smc.gz', 'CNS.CNN.10.smc.gz', 'CNS.CNN.11.smc.gz', 'CNS.CNN.12.smc.gz', 'CNS.CNN.13.smc.gz', 'CNS.CNN.14.smc.gz', 'CNS.CNN.15.smc.gz', 'CNS.CNN.16.smc.gz', 'CNS.CNN.17.smc.gz', 'CNS.CNN.18.smc.gz', 'CNS.CNN.19.smc.gz', 'CNS.CNN.1.smc.gz', 'CNS.CNN.20.smc.gz', 'CNS.CNN.21.smc.gz', 'CNS.CNN.22.smc.gz', 'CNS.CNN.23.smc.gz', 'CNS.CNN.24.smc.gz', 'CNS.CNN.25.smc.gz', 'CNS.CNN.26.smc.gz', 'CNS.CNN.27.smc.gz', 'CNS.CNN.28.smc.gz', 'CNS.CNN.29.smc.gz', 'CNS.CNN.2.smc.gz', 'CNS.CNN.3.smc.gz', 'CNS.CNN.4.smc.gz', 'CNS.CNN.5.smc.gz', 'CNS.CNN.6.smc.gz', 'CNS.CNN.7.smc.gz', 'CNS.CNN.8.smc.gz', 'CNS.CNN.9.smc.gz'], em_iterations=20, ftol=0.0001, length_cutoff=10000, no_filter=False, no_initialize=False, nonseg_cutoff=None, outdir='CNN_CNS', polarization_error=0.5, pop1='../CNN/CNN/model.final.json', pop2='../CNS/CNS/model.final.json', regularization_degree=2, regularization_penalty=7, thinning=None, unfold=False, verbose=0, xtol=0.1)
549 smcpp.analysis DEBUG Polarization error p=0.500000
549 smcpp.analysis DEBUG regularization degree=2
549 smcpp.analysis INFO Loading data...
13600 smcpp.analysis INFO 5.02 Gb of data
13600 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.13.smc.gz', n=array([36, 40]), a=array([2, 0]))
13601 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.11.smc.gz', n=array([38, 38]), a=array([2, 0]))
13601 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.6.smc.gz', n=array([38, 38]), a=array([2, 0]))
13602 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.9.smc.gz', n=array([38, 38]), a=array([2, 0]))
13602 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.27.smc.gz', n=array([36, 40]), a=array([2, 0]))
13603 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.17.smc.gz', n=array([38, 38]), a=array([2, 0]))
13603 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.4.smc.gz', n=array([36, 40]), a=array([2, 0]))
13604 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.12.smc.gz', n=array([36, 40]), a=array([2, 0]))
13604 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.21.smc.gz', n=array([36, 40]), a=array([2, 0]))
13604 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.26.smc.gz', n=array([36, 40]), a=array([2, 0]))
13605 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.14.smc.gz', n=array([36, 40]), a=array([2, 0]))
13605 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.29.smc.gz', n=array([38, 38]), a=array([2, 0]))
13606 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.3.smc.gz', n=array([36, 40]), a=array([2, 0]))
13606 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.7.smc.gz', n=array([36, 40]), a=array([2, 0]))
13606 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.5.smc.gz', n=array([36, 40]), a=array([2, 0]))
13607 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.27.smc.gz', n=array([38, 38]), a=array([2, 0]))
13607 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.26.smc.gz', n=array([38, 38]), a=array([2, 0]))
13608 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.21.smc.gz', n=array([38, 38]), a=array([2, 0]))
13608 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.28.smc.gz', n=array([38, 38]), a=array([2, 0]))
13608 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.7.smc.gz', n=array([38, 38]), a=array([2, 0]))
13609 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.29.smc.gz', n=array([36, 40]), a=array([2, 0]))
13609 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.2.smc.gz', n=array([36, 40]), a=array([2, 0]))
13610 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.6.smc.gz', n=array([36, 40]), a=array([2, 0]))
13610 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.5.smc.gz', n=array([38, 38]), a=array([2, 0]))
13610 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.20.smc.gz', n=array([36, 40]), a=array([2, 0]))
13611 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.20.smc.gz', n=array([38, 38]), a=array([2, 0]))
13611 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.16.smc.gz', n=array([38, 38]), a=array([2, 0]))
13612 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.11.smc.gz', n=array([36, 40]), a=array([2, 0]))
13612 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.23.smc.gz', n=array([38, 38]), a=array([2, 0]))
13612 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.19.smc.gz', n=array([36, 40]), a=array([2, 0]))
13613 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.23.smc.gz', n=array([36, 40]), a=array([2, 0]))
13613 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.2.smc.gz', n=array([38, 38]), a=array([2, 0]))
13614 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.25.smc.gz', n=array([38, 38]), a=array([2, 0]))
13614 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.16.smc.gz', n=array([36, 40]), a=array([2, 0]))
13614 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.28.smc.gz', n=array([36, 40]), a=array([2, 0]))
13615 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.18.smc.gz', n=array([38, 38]), a=array([2, 0]))
13615 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.15.smc.gz', n=array([36, 40]), a=array([2, 0]))
13616 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.18.smc.gz', n=array([36, 40]), a=array([2, 0]))
13616 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.14.smc.gz', n=array([38, 38]), a=array([2, 0]))
13616 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.22.smc.gz', n=array([36, 40]), a=array([2, 0]))
13617 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.19.smc.gz', n=array([38, 38]), a=array([2, 0]))
13617 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.13.smc.gz', n=array([38, 38]), a=array([2, 0]))
13617 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.17.smc.gz', n=array([36, 40]), a=array([2, 0]))
13618 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.25.smc.gz', n=array([36, 40]), a=array([2, 0]))
13618 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.4.smc.gz', n=array([38, 38]), a=array([2, 0]))
13619 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.24.smc.gz', n=array([38, 38]), a=array([2, 0]))
13619 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.12.smc.gz', n=array([38, 38]), a=array([2, 0]))
13619 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.1.smc.gz', n=array([38, 38]), a=array([2, 0]))
13620 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.15.smc.gz', n=array([38, 38]), a=array([2, 0]))
13620 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.22.smc.gz', n=array([38, 38]), a=array([2, 0]))
13621 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.24.smc.gz', n=array([36, 40]), a=array([2, 0]))
13621 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.8.smc.gz', n=array([36, 40]), a=array([2, 0]))
13621 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.9.smc.gz', n=array([36, 40]), a=array([2, 0]))
13622 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.8.smc.gz', n=array([38, 38]), a=array([2, 0]))
13622 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.3.smc.gz', n=array([38, 38]), a=array([2, 0]))
13623 smcpp.analysis DEBUG Contig(pid=('CNN', 'CNS'), fn='CNN_CNS.10.smc.gz', n=array([38, 38]), a=array([2, 0]))
13623 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.1.smc.gz', n=array([36, 40]), a=array([2, 0]))
13623 smcpp.analysis DEBUG Contig(pid=('CNS', 'CNN'), fn='CNS.CNN.10.smc.gz', n=array([36, 40]), a=array([2, 0]))
13624 smcpp.analysis INFO 2 populations
49195 smcpp.analysis DEBUG Empirical CSFS:
0.0
```
Thanks for your help.

Best
Zhuqing

**Comment by terhorst** [comment]
Can you share one of the ``CNS.CNN.x.smc.gz`` files? For some reason I do not encounter this error on my test data sets. Thanks!

**Comment by biozzq** [comment]
Dear @terhorst 

I attached my files for you, the Nemodel.zip includes the marginal estimates  produced by `estimate`. Thanks for your time. 
[CNN.CNS.29.smc.gz](https://github.com/popgenmethods/smcpp/files/1007521/CNN.CNS.29.smc.gz)
[CNS.CNN.29.smc.gz](https://github.com/popgenmethods/smcpp/files/1007523/CNS.CNN.29.smc.gz)
[Nemodel.zip](https://github.com/popgenmethods/smcpp/files/1007525/Nemodel.zip)

Best
Zhuqing

**Comment by terhorst** [comment]
Thanks for uploading the files. I believe this is finally fixed in 1.9.2.
